add CI and BUILD_NUMBER env var in docs (#10749)
pytest-dev/pytest: changelog/10690.doc.rst

> Add CI and BUILD_NUMBER environment variables in documentation

Reviewer: ```suggestion
Added `CI` and `BUILD_NUMBER` environment variables to the documentation.
```
